### PR TITLE
Remove dead code in induction internals.

### PR DESCRIPTION
**Pull request by ppedrot**
It was known statically that ElimOver nodes could only be built with the recursive flag set to true.

### CODE REVIEW

File: tactics/induction.ml [hunk 1]
```diff
@@ -1031,7 +1031,7 @@ let find_induction_type env sigma isrec elim hyp0 sort = match elim with
   let is_elim, ind, typ = guess_elim_shape env sigma isrec sort hyp0 in
   (* We drop the scheme and elimc/elimt waiting to know if it is dependent, this
     needs no update to sigma at this point. *)
-  let elim = if is_elim then ElimOver (isrec, hyp0, ind) else CaseOver (hyp0, ind) in
+  let elim = if is_elim then ElimOver (hyp0, ind) else CaseOver (hyp0, ind) in
```
Comment (SkySkimmer) [inline tactics/induction.ml:1034]: I don't see how isrec is statically known to be true here

Comment (SkySkimmer) [reply tactics/induction.ml:1034]: Then how about building the `elim` there instead of returning `is_elim`

Comment (ppedrot) [reply tactics/induction.ml:1034]: I want to keep the code separate, because eventually CaseOver will be a fully different tactic.

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**